Failure of `test_surface.py::test_sampling_affine` using pre-release dependencies

Seen on this workflow.

https://github.com/nilearn/nilearn/actions/runs/8349027568/job/22852309799?pr=4289

This is a new failure of this test:

https://github.com/nilearn/nilearn/blob/4691c77ae006193f69123345d9f8480ef5a265a1/nilearn/surface/tests/test_surface.py#L671

To reproduce:

```python
tox run -e test_pre -- nilearn/surface/tests/test_surface.py::test_sampling_affine
```

Pytest failure

```
args = (<built-in function eq>, array([1., 2., 1.]), [1.0, 2.0, 1.0])
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 3 / 3 (100%)
E           Max absolute difference: 4.4408921e-16
E           Max relative difference: 2.22044605e-16
E            x: array([1., 2., 1.])
E            y: array([1., 2., 1.])

args       = (<built-in function eq>, array([1., 2., 1.]), [1.0, 2.0, 1.0])
func       = <function assert_array_compare at 0x7ca53c64a980>
kwds       = {'err_msg': '',
 'header': 'Arrays are not equal',
 'strict': False,
 'verbose': True}
self       = <contextlib._GeneratorContextManager object at 0x7ca53c670810>
```

--- a/nilearn/surface/tests/test_surface.py
+++ b/nilearn/surface/tests/test_surface.py
@@ -678,10 +678,10 @@ def test_sampling_affine(affine_eye):
     affine[-1, -1] = 1
     texture = surface._nearest_voxel_sampling(
         [img], mesh, affine=affine, radius=1, kind='ball')
-    assert_array_equal(texture[0], [1., 2., 1.])
+    assert_array_almost_equal(texture[0], [1.0, 2.0, 1.0], decimal=15)
     texture = surface._interpolation_sampling(
         [img], mesh, affine=affine, radius=0, kind='ball')
-    assert_array_almost_equal(texture[0], [1.1, 2., 1.])
+    assert_array_almost_equal(texture[0], [1.1, 2.0, 1.0], decimal=15)
 
 
 @pytest.mark.parametrize("kind", ["auto", "line", "ball"])

